### PR TITLE
Add gitops-promotion Workflow Template

### PR DESCRIPTION
**Pull request by TedSpinks**

### CODE REVIEW

File: workflows/gitops-promotion/versions/0.0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# gitops-promotion
+
+## Summary
+
+Templates for doing GitOps-style promotions of microservices from one K8s environment to another. Assumes that you have a GitOps monorepo containing subdirectories that represent environments, each of which contains the manifests for deploying that environment to a K8s cluster. Supported manifest types include Kustomize, Helm Charts, and raw YAML files.
+
+## Templates
+
+1. [promote-from-src-to-dest-env](./docs/promote-from-src-to-dest-env.md)
```
Comment (eti-codefresh) [inline workflows/gitops-promotion/versions/0.0.1/README.md:9]: please give the full path. it's not working well when accessing it from argo hub

File: workflows/gitops-promotion/versions/0.0.1/workflowTemplate.yaml [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+    argo-hub/version: '0.0.1'
+    argo-hub/description: 'Templates to promote microservices between K8s environments within your GitOps monorepo'
+    argo-hub/categories: 'gitops,git,utilities,deployment'
+    argo-hub/license: 'MIT'
+    argo-hub/owner_name: 'Ted Spinks'
+    argo-hub/owner_email: 'ted.spinks@codefresh.io'
+    argo-hub/owner_avatar: 'https://avatars.githubusercontent.com/TedSpinks'
+    argo-hub/owner_url: 'https://github.com/TedSpinks'
+    argo-hub/icon_url: "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/workflows/gitops-promotion/assets/icon.svg"
+    argo-hub/icon-background: "#f4f4f4"
```
Comment (eti-codefresh) [inline workflows/gitops-promotion/versions/0.0.1/workflowTemplate.yaml:15]: argo-hub/icon_background

File: workflows/gitops-promotion/versions/0.0.1/workflowTemplate.yaml [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+          - name: helm-dep-pattern    # For helm-dependency - name of the subchart/dependency to copy from source to dest
+            value: "[[SVC_NAME]]"
+      outputs:
+        parameters:
+          - name: codefresh-io-pr-url
+            valueFrom:
+              default: ""
+              path: /tmp/pr-url.txt
+      container:
+        image: quay.io/codefreshplugins/argo-hub-workflows-gitops-promotion-versions-0.0.1-images-gitops-promotion:main
```
Comment (eti-codefresh) [inline workflows/gitops-promotion/versions/0.0.1/workflowTemplate.yaml:72]: this image should be public

**Review by eti-codefresh — APPROVED** [review]

File: workflows/gitops-promotion/versions/0.0.1/README.md (hunk 1 above)
Comment (TedSpinks) [reply workflows/gitops-promotion/versions/0.0.1/README.md:9]: Will do!

File: workflows/gitops-promotion/versions/0.0.1/workflowTemplate.yaml (hunk 2 above)
Comment (TedSpinks) [reply workflows/gitops-promotion/versions/0.0.1/workflowTemplate.yaml:15]: Got it!

File: workflows/gitops-promotion/versions/0.0.1/workflowTemplate.yaml (hunk 3 above)
Comment (TedSpinks) [reply workflows/gitops-promotion/versions/0.0.1/workflowTemplate.yaml:72]: I'll ask DevOps to make it public as soon as this merges.

**Pull request merged**